micromanager_gui/main_window.py

```diff
@@ -147,14 +149,156 @@ def add_frame_explorer(self, name, array):
+                
+                #xypositions
+                if event_index_p > 0 and event_index_c == 0 and event_index_z == 0 and event_index_t == 0:
+                    empty_im = np.empty(((1,)*(len(sequence.axis_order)-2) + (z_stack_length,c_stack_length,) + image.shape), dtype=np.uint16)
+                    layer.data = np.concatenate((layer.data,empty_im), axis=-5)
+
+                #timepoints
+                if event_index_t > 0 and event_index_z == 0 and event_index_c == 0 and event_index_p == 0:
+                    empty_im = np.empty(((1,p_stack_length,z_stack_length,c_stack_length,) + image.shape), dtype=np.uint16)
+                    layer.data = np.concatenate((layer.data,empty_im), axis=-6)
+
+
+            if sequence.axis_order == 'tpcz' or sequence.axis_order == 'ptcz':
+                #zpositions
+                if event_index_z > 0 and event_index_c == 0 and event_index_p == 0 and event_index_t == 0:
+                    empty_im = np.empty(((1,)*len(sequence.axis_order) + image.shape), dtype=np.uint16) 
+                    layer.data = np.concatenate((layer.data,empty_im), axis=-4)
+                
+                #channels
+                if event_index_c > 0 and event_index_z == 0 and event_index_p == 0 and event_index_t == 0:
+                    empty_im = np.empty(((1,)*(len(sequence.axis_order)-2) + (z_stack_length,1,) + image.shape), dtype=np.uint16)
+                    layer.data = np.concatenate((layer.data,empty_im), axis=-3)
+
+                #xypositions
+                if event_index_p > 0 and event_index_c == 0 and event_index_z == 0 and event_index_t == 0:
+                    empty_im = np.empty(((1,)*(len(sequence.axis_order)-2) + (z_stack_length,c_stack_length,) + image.shape), dtype=np.uint16)
+                    layer.data = np.concatenate((layer.data,empty_im), axis=-5)
+
+                #timepoints
+                if event_index_t > 0 and event_index_z == 0 and event_index_c == 0 and event_index_p == 0:
+                    empty_im = np.empty(((1,p_stack_length,z_stack_length,c_stack_length,) + image.shape), dtype=np.uint16)
+                    layer.data = np.concatenate((layer.data,empty_im), axis=-6)
+
+            layer.data[event_index_t,event_index_p,event_index_z,event_index_c, ...] = image
+         
+            #set which dimension to display in napari viewer -> viewer.dims.set_point(axis, index)
+            self.viewer.dims.set_point(3, event_index_c)
+            self.viewer.dims.set_point(2, event_index_z)
+            self.viewer.dims.set_point(1, event_index_p)
+            self.viewer.dims.set_point(0, event_index_t)   
+
         except KeyError:
-            self.viewer.add_image(img, name=name)
+
+            layer = self.viewer.add_image(image[(np.newaxis,)*len(sequence.axis_order)], name=layer_name)
+            
+            self.viewer.dims.axis_labels = 'tpzcyx'
+
+            layer.metadata['Metadata'] = sequence
```

Comment:
```suggestion
            layer.metadata['useq_sequence'] = sequence
```

(`metadata['metadata']` seems a big vague)